src/BugsnagUnity/Payload/Metadata.cs

```diff
@@ -32,61 +33,45 @@ public void AddMetadata(string section, IDictionary<string, object> metadataSect
             }
             if (SectionExists(section))
             {
-
                 var existingSection = (IDictionary<string, object>)Get(section);
 
-                foreach (var entry in metadataSection)
+                foreach (var key in metadataSection.Keys.ToList())
```

Comment:
Is there an advantage to using the keys directly rather than an entryset?

Reply:
It allows for the set to be edited while iterating